### PR TITLE
Formatted endpoints

### PR DESCRIPTION
**Pull request by aviraljain99**
Please ignore the change in the dockerfile. I will remove it in the next commit to this PR - it was just for testing. But I have formatted the endpoints directory using black and that is the subject of this PR.

### CODE REVIEW

**Comment by aviraljain99** [comment]
I was also wondering if any of you had any thoughts on the maximum line length that is enforced by black? The black default is 88 characters.

**Comment by nicklambourne** [comment]
I'm in favour of 100 char line lengths, 120 at most.

**Review by nicklambourne — COMMENTED** [review]
Was `black` already in the dependcy closure in pyproject.toml?  What's the process for running black in the repo? Where is the config set?

**Comment by aviraljain99** [comment]
@nicklambourne I don't think the black formatter is setup within this repo - neither is it in the pyproject.toml dependency section (is it referred to as a closure?). I will look into it - also sorry for the late response.

File: .github/workflows/black_python_formatter.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+      ###################################
+      # Run Formatter against code base #
+      ###################################
+      - name: Format Code Base
+        uses: github/super-linter@v4
+        env:
+          VALIDATE_ALL_CODEBASE: false
+          VALIDATE_PYTHON_BLACK: true
+          DEFAULT_BRANCH: dev
```
Comment (harrykeightley) [inline .github/workflows/black_python_formatter.yml:41]: Is this meant to be dev?

**Comment by aviraljain99** [comment]
@nicklambourne I have copied over the workflow from elpiscloud that runs the black formatter on the Python files so it should now run that workflow on pushes to master or when there is a PR into master. 

I also added some black configuration info to the pyproject.toml file and set the line length to a 100 characters. Also black is now a dev dependency. Is this PR good to merge into master now?

File: .github/workflows/black_python_formatter.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      ###################################
+      # Run Formatter against code base #
+      ###################################
+      - name: Format Code Base
+        uses: github/super-linter@v4
+        env:
+          VALIDATE_ALL_CODEBASE: false
+          VALIDATE_PYTHON_BLACK: true
+          DEFAULT_BRANCH: master
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (nicklambourne) [inline .github/workflows/black_python_formatter.yml:42]: Missing newline

File: .github/workflows/black_python_formatter.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    # Name the Job
+    name: Formatter
+    # Set the agent to run on
+    runs-on: ubuntu-latest
```
Comment (nicklambourne) [inline .github/workflows/black_python_formatter.yml:18]: This should be pinned at a specific version so it doesn't break without warning on us at some undefined point in the future when the definition of what constitutes `latest` changes.

File: .github/workflows/black_python_formatter.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    # Name the Job
+    name: Formatter
```
Comment (nicklambourne) [inline .github/workflows/black_python_formatter.yml:16]: May want to be more specific, as we'll no doubt have linters for other languages (TS, even HCL)

File: pyproject.toml [hunk 5]
```diff
@@ -31,6 +31,11 @@ psutil = "^5.8.0"
 pytest = "^6.2.1"
 pylint = "^2.6.0"
 wheel = "^0.36.2"
+black = "^22.3.0"
+
+[tool.black]
+line-length = 100
+target-version = ['py38']
```
Comment (nicklambourne) [inline pyproject.toml:38]: Are we limiting ourselves to py38 for some particular reason?

**Review by nicklambourne — CHANGES_REQUESTED** [review]
Few nits etc. to address

**Review by nicklambourne — APPROVED** [review]

**Pull request merged**